sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/changefeed/implementation/AutoCheckpointer.java

```diff
@@ -50,6 +53,10 @@ public void close(ChangeFeedObserverContext context, ChangeFeedObserverCloseReas
     @Override
     public Mono<Void> processChanges(ChangeFeedObserverContext context, List<CosmosItemProperties> docs) {
         return this.observer.processChanges(context, docs)
+            .onErrorResume(throwable -> {
+                logger.warn("Unexpected exception from thread {}", Thread.currentThread().getId(), throwable);
```

Comment:
logger.error ?
logger.debug ?

Reply:
logger.warn is the right thing here; the exception could be thrown from the user code which handles the feeds. In any case the side effect of this is to kill the current worker processing the feeds and start a new one from where we left (some documents might be processed multiple times).